src/main/java/seedu/address/model/task/exceptions/TaskNotFoundException.java

```diff
@@ -0,0 +1,6 @@
+package seedu.address.model.task.exceptions;
+
+/**
+ * Signals that the operation is unable to find the specified task;
+ */
+public class TaskNotFoundException extends RuntimeException {}
```

Comment:
Perhaps you can consider adding an error message for this exception so exception handling would be clearer for us and users will be aware of the incorrect or invalid command provided.

Reply:
TaskNotFoundException is an unchecked exception since it inherits from RunTimeException. I think the purpose of this exception is mainly is for bug checking. In addition, the error message that will be shown to the user is handled in the each of the Command class instead. 

For instance, below code is taken from EditCommand class from the original AB-3 (this is absent from our current EditModuleCommand class though):

"if (index.getZeroBased() >= lastShownList.size()) {
    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
}

However, I think we can still add error message just to make it easier for the programmer to understand the error when debugging 😃

Reply:
Yes, that sounds fine.